core/src/main/javaTemplate/org/python/core/PyLongMethods.java

```diff
@@ -0,0 +1,81 @@
+package org.python.core;
+
+import java.math.BigInteger;
+import static java.math.BigInteger.ZERO;
+import static java.math.BigInteger.ONE;
+
+import org.python.core.PyObjectUtil.NoConversion;
+
+// $OBJECT_GENERATOR$ PyLongGenerator
+
+/**
+ * This class contains static methods implementing operations on the
+ * Python {@code int} object, supplementary to those defined in
+ * {@link PyLong}.
+ * <p>
+ * These methods may cause creation of descriptors in the dictionary of
+ * the type. Those with reserved names in the data model will also fill
+ * slots in the {@code Operations} object for the type.
+ * <p>
+ * Implementations of binary operations defined here will have
+ * {@code Object} as their second argument, and should return
+ * {@link Py#NotImplemented} when the type in that position is not
+ * supported.
+ */
+class PyLongMethods {
+
+    private PyLongMethods() {}  // no instances
+
+    // $SPECIAL_METHODS$ ---------------------------------------------
+
+    // plumbing ------------------------------------------------------
+
+    /**
+     * Convert an {@code int} or its sub-class to a Java
+     * {@code BigInteger}. Conversion may raise an exception that is
+     * propagated to the caller. If the Java type of the {@code int} is
+     * declared, generally there is a better option than this method. We
+     * only use it for {@code Object} arguments. If the method throws
+     * the special exception {@link NoConversion}, the caller must catch
+     * it, and will normally return {@link Py#NotImplemented}.
+     * 
+     * @param v to convert
+     * @return converted to {@code BigInteger}
+     * @throws NoConversion v is not an {@code int}
+     */
+    private static BigInteger toBig(Object v) throws NoConversion {
+        // Check against supported types, most likely first
+        if (v instanceof Integer)
+            return BigInteger.valueOf(((Integer) v).longValue());
+        else if (v instanceof BigInteger)
+            return (BigInteger) v;
+        else if (v instanceof PyLong)
+            return ((PyLong) v).value;
+        else if (v instanceof Boolean)
+            return (Boolean) v ? ONE : ZERO;
+
+        throw PyObjectUtil.NO_CONVERSION;
+    }
+
+    /**
+     * Reduce a {@code BigInteger} result to {@code Integer} if
+     * possible. This makes it more likely the next operation will be
+     * 32-bit.
+     * 
+     * @param r to reduce
+     * @return equal value
+     */
+    static Object toInt(BigInteger r) {
+        /*
+         * Implementation note: r.intValueExact() is for exactly this
```

Comment:
Maybe a use case for [@implNote](https://openjdk.java.net/jeps/8068562). Despite the draft status the described tags are already in frequent use in current and recent Java standard API doc since several years.

Reply:
You're right and I've used it elsewhere in the code. It needs some [special pleading in the javadoc command](https://github.com/jeff5/jython/blob/8fcaec5b0d3ce4bd7bf39d6cc77e9a6f57f5b515/core/core.gradle#L99-L108) to enable it. Probably, this text is from before I learned that.

Reply:
Actually, in context, I see this is regular comment, not Javadoc. It is definitely a note to maintainers, not to implementers of an API, explaining why I did not use the provided method. There are probably bits of Javadoc that could do with `@implNote` but none so obvious. The JEP is a good reminder how to use the tags.